packages/nimble-components/src/wafer-map/specs/features/rendering.md

```diff
@@ -0,0 +1,207 @@
+
+The warning will be displayed by the existing warning signaling mechanism used in the Ozone application.
+![warning display](resources/warning-integration.png)
+
+### Performance Testing
+
+We wish to measure functions performance of wafer map using the existing lighthouse setup paired with windows.performance tools.
+
+Example:
+
+```TS
+    const functionName = 'createWaferMap';
+    performance.mark(`${functionName}_start`);
+    const wafermap = document.createElement('nimble-wafer-map');
+    document.body.appendChild(wafermap);
+    performance.mark(`${functionName}_end`);
+    performance.measure(functionName, `${functionName}_start`, `${functionName}_end`);
+```
+
+The example shows how custom markers can be set in the tested code. The result of these user marks can be observed inside lighthouse like in the screenshot bellow.
+
+![User Timing and measures](./resources/user-timing-marks-and-measures.PNG)
+
+We wish to extend the existing lighthouse CI testing to include the wafermap component and record various metrics about the performance it has when rendering multiple generated data loads. They will include data ingestion, rendering time, rerendering when user events get triggered and data changes.
+
+These records will be made in a separate web page from the existing performance tests to avoid polluting the results. They will not be preserved over time, each CI record being available for download after build.
+
+Our target goals are:
+
+1. For wafers >200k and <1m die
+
+    1. Initial rendering: <1 second
+    1. Zoom/pan actions: <500ms
+    1. Configuration changes that don't modify the dataset (e.g. show labels):
+       <500ms
+    1. Configuration changes that modify the dataset (e.g. replace layers): <1 second
+
+1. For wafers >1m and <10m die
+    1. Initial rendering: <10 seconds
+    1. Zoom/pan actions: <1 second
+    1. Interactions that don't modify the dataset (e.g. show labels): <1 second
+    1. Interactions that modify the dataset (e.g. replace layers): <10 seconds
+
+## Alternative Implementations / Designs
+
+### Alternative Data Structures and Interfaces
+
+The above mentioned structure can also be implemented as an [apache arrow](https://arrow.apache.org/docs/js/index.html) table with columns and metadata. This may introduce an additional overhead that is not needed for the moment as the research towards database caching is still proceeding.
+
+Another option is to break each object property as a separate attribute for the wafer map component. This can also lead to increased complexity and confusion for the user which will need to pass several structured objects instead of a singular object.
+
+### Alternative Rendering
```

Comment:
I think this could be a great ideea. The wafer maps may have some different interaction feature sets, like grids, tooltips, and others, but there is an overlap. It may be something worth looking into in the future. Right now, there are still developments for single image layer wafer galleries and support for large images is being implemented, but this is a direction that could be explored.